### ISSUE TITLE
non-exhaustive patterns false positive

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#73919]
```rust
fn foo(x: usize) -> u32 {
    match x {
        0 ..= usize::MAX => 0
    }
}
fn main() {}
```
Gives:
```
error[E0004]: non-exhaustive patterns: `_` not covered
 --> ...\test.rs:2:11
  |
2 |     match x {
  |           ^ pattern `_` not covered
  |
  = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
  = note: the matched value is of type `usize`
```

**Comment by jonas-schievink** [comment]
Fairly sure this is intentional for `usize`

**Comment by leonardo-m** [comment]
From the pull 73937 I've seen a part that I didn't see/remember:

>An unstable feature precise_pointer_size_matching will be added to permit matching exactly on pointer-size integer types.

So you're right, this isn't a new bug, it's a known unfinished part. I hope to see that precise_pointer_size_matching implemented.

**Comment by varkor** [comment]
> I hope to see that precise_pointer_size_matching implemented.

It already exists.

**Comment by leonardo-m** [comment]
I see, then what do I have to hope to see? :-)

**Comment by varkor** [comment]
I'm not sure the feature is ever going to be stabilised, because there are concerns that this isn't a desirable feature. But hopefully the diagnostics will at least be improved in https://github.com/rust-lang/rust/pull/73937.

**Comment by leonardo-m** [comment]
Do you have a link to such concerns? (Or are you willing to summarize them?)
(I value your work on the compiler a lot varkor :-) )

**Comment by varkor** [comment]
I think the relevant concerns arose in the [RFC thread](https://github.com/rust-lang/rfcs/pull/2591#issuecomment-438697831) (from this comment up). The overview is that we ought not to make assumptions about pointer sizes on the target platform, because they're not guaranteed to be fixed, so this could cause compatibility issues.
